### ISSUE TITLE
Investigate methods to reduce cf-d deployment time

### ISSUE THREAD

**Issue by mogul** [cloud-gov/cg-product#991]

**Comment by RonWilliams** [comment]
Per @jontours -- this should be lower priority as the deployment is stable even though it takes a long time to complete.

**Comment by karareinsel** [comment]
This story needs to be further fleshed out to be considered for an upcoming sprint.

**Comment by karareinsel** [comment]
It takes 2-4 hours to deploy cf, and there are non-breaking changes we can make to our deployment strategy that will decrease deployment times. Lower priority.

**Comment by karareinsel** [comment]
@mxplusb will add his research findings here.

**Comment by mxplusb** [comment]
Okay, so it turns out there is a way to set a per-instance group update block which sets the pace for updates within that instance group.

I'd like to update cf-d in production using an update block specific to the diego cells which does a `create-swap-delete` on 33% of the diego cells. That's the slowest part of deployments and decreasing that instance group deployment time should bring down cf deployments to a much more reasonable time. We felt this most painfully on the k8s recursor issue, where apps had to get the updated DNS config value, so we had to go into every single stemcell and update it ahead of time while we waiting on cf to deploy. Being able to push changes out faster to diego will help prevent stuff like that in the future.

**Comment by karareinsel** [comment]
Tackle as part of pipeline refactoring.
